src/java.desktop/share/classes/sun/java2d/pipe/RenderBuffer.java

```diff
@@ -170,9 +170,9 @@ public RenderBuffer put(short[] x) {
     public RenderBuffer put(short[] x, int offset, int length) {
         // assert (position() % SIZEOF_SHORT == 0);
         if (length > COPY_FROM_ARRAY_THRESHOLD) {
-            long offsetInBytes = offset * SIZEOF_SHORT + Unsafe.ARRAY_SHORT_BASE_OFFSET;
-            long lengthInBytes = length * SIZEOF_SHORT;
-            unsafe.copyMemory(x, offsetInBytes, null, curAddress, lengthInBytes);
+            int offsetInBytes = offset * SIZEOF_SHORT;
+            int lengthInBytes = length * SIZEOF_SHORT;
+            MemorySegment.copy(x, offsetInBytes, segment, JAVA_SHORT, curOffset, length);
```

Comment:
This doesn't look right. When copying from an array like this, the source offset is an array index, not an offset in bytes. So I believe the second argument here should just be `offset`.

Reply:
Ah. I was using the copy method that uses a MemorySegment wrapped array then my microbenchmark showed the unwrapped one was  faster, so I made a late switch (although then did a full retest) but missed that.

I think testing is still fine because offset is always zero in our code .. 

I will fix that, and think about the other comments.